README.md

```diff
@@ -1 +1,475 @@
+  - [Search Google Images](#search-google-images)
+- [Advanced search API usage](#advanced-search-api-usage)
+  - [SerpApi client](#serpapi-client)
+  - [Batch Asynchronous client](#batch-asynchronous-client)
+- [Change log](#change-log)
+- [Roadmap](#roadmap)
+- [Conclusion](#conclusion)
+- [Contributing](#contributing)
+
+## API Guide
+
+### Search API overview
+```ruby
+# initialize the client
+client = SerpApi::Client.new(api_key: "private key")
+
+# search query
+query = {
+  engine: "google", # full list: https://serpapi.com/search-api
+  q: "client",
+  google_domain: "Google Domain",
+  location: "Location Requested", # example: Portland,Oregon,United States
+  device: "desktop|mobile|tablet",
+  hl: "Google UI Language",
+  gl: "Google Country",
+  safe: "Safe Search Flag",
+  num: "Number of Results",
+  start: "Pagination Offset",
+  tbm: "nws|isch|shop",
+  tbs: "custom to be client criteria"
+  async: true|false # allow async call
+}
+
+# formated search results as a Hash
+#  serpapi.com converts HTML -> JSON 
+results = client.search(query)
+
+# raw html as a String
+#  serpapi.com acts a proxy to provive high throughputs
+doc = client.html(query)
+```
+
+(The full documentation)[https://serpapi.com/client-api].
+More hands on examples are available belows.
+
+
+### Location API
+
+```ruby
+client = SerpApi::Client.new() 
+location_list = client.location(q: "Austin", limit: 3)
+puts "number of location: #{location_list.size}"
+pp location_list
+```
+
+it prints the first 3 location matching Austin (Texas, Texas, Rochester)
+```ruby
+[{:id=>"585069bdee19ad271e9bc072",
+  :google_id=>200635,
+  :google_parent_id=>21176,
+  :name=>"Austin, TX",
+  :canonical_name=>"Austin,TX,Texas,United States",
+  :country_code=>"US",
+  :target_type=>"DMA Region",
+  :reach=>5560000,
+  :gps=>[-97.7430608, 30.267153],
+  :keys=>["austin", "tx", "texas", "united", "states"]},
+  ...]
+```
+
+### Search Archive API
+
+This API allows to retrieve previous client.
+To do so run a client to save a search_id.
+```ruby
+client = SerpApi::Client.new(api_key: ENV['API_KEY'], engine: 'google')
+original_search = client.search(q: "Coffee", location: "Portland")
+search_id = original_search[:search_metadata][:id]
+```
+
+Now let retrieve the previous client from the archive.
+
+```ruby
+client = SerpApi::Client.new(api_key: ENV['API_KEY'])
+archive_results = client.search_archive(search_id)
+pp archive_results
+```
+it prints the search results from archive.
+
+### Account API
+```ruby
+client = SerpApi::Client.new(api_key: ENV['API_KEY'])
+pp client.account
+```
+it prints your account information.
+
+## Basic example per search engine
+### Search Google Images
+```ruby
+require 'serpapi'
+client = SerpApi::Client.new(tbm: 'isch', api_key: ENV['API_KEY'], engine: 'google')
+image_results_list = client.search(q: 'coffee')[:images_results]
+image_results_list[0..2].each do |image_result|
+puts image_result[:original]
```

Comment:
```suggestion
image_results_list[0..2].each do |image_result|
  puts image_result[:original]
end
```

Reply:
code update to `pp results[:images_results]`. so it's more generic for all the templated spec files